bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolManager.java

```diff
@@ -225,8 +225,9 @@ public void schedule(final Runnable task, final Expression expression) {
 
         @Override
         public boolean remove(Runnable task) {
-            if (futures.get(task) != null && futures.get(task).cancel(false)) {
-                running.remove(task);
+            Future<?> future = futures.get(task);
+            if (future != null && future.cancel(false)) {
+                running.remove(future);
```

Comment:
It is not me who is confused, it is the JVM. The current implementation simply does not work, as `running` simply does not contain the instances that you try to remove. This PR fixes this.